data/src/main/java/org/apache/iceberg/data/BaseFileWriterFactory.java

```diff
@@ -118,13 +117,15 @@ public DataWriter<T> newDataWriter(
 
         case PARQUET:
           Parquet.DataWriteBuilder parquetBuilder =
-              Parquet.writeData(outputFile)
+              Parquet.writeData(file.rawOutputFile())
```

Comment:
This is a behavior change. I think we need to check whether `keyMetadata` is a `KeyMetadata` in order to do this. If it's metadata that can be used for Parquet native encryption then we can use it. Otherwise we should fall back to using `encryptingOutputFile()` like before.